virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,151 @@
+import os
+import os.path
+import pickle
+import logging
+
+from avocado.utils import process
+from avocado.core import exceptions
+
+"""Handle Linux kernel modules"""
+
+
+class KernelModuleHandler(object):
+    """Class handling Linux kernel modules"""
+
+    def __init__(self, module_name, persistence_file=""):
+        """
+        Create kernel module handler. It backs up the current status
+        of the kernel module or get its current status from a backed up
+        file.
+
+        :param module_name:
+        :param persistence_file: absolute file path to store object in
+        """
+        self._module_name = module_name
+        self._was_loaded = None
+        self._loaded_config = ""
```

Comment:
I know in C mixing strings and Nonetype is not recommended (and hard), but in python it's the usual way to distinguish between not defined and empty string. I don't say it's a mistake, just a different from the usual...